### PR TITLE
Cleanup legacy code and minor refactor

### PR DESCRIPTION
**Pull request by zheileman**
https://trello.com/c/KfT2DRuE

Part of the work done to cleanup legacy platform/code from services where we still needed to handle legacy.

Mostly deleting no longer needed code, and minor renaming/refactoring where it makes sense.
Ensuring we keep coverage at 100%.

Minor bump to gems, removing notify client.

### CODE REVIEW

File: app/value_objects/raw_message.rb
Comment (zheileman) [inline app/value_objects/raw_message.rb:1]: I just took out V2::RawMessage from its V2 namespace, replacing (legacy) RawMessage, and it looks like code changes but is just deleting old file, replacing with new one. Some code is the same.
There is some inconsistency in casing tho with the headers but I've not changed anything.

File: lib/tasks/replay_submission.rake [hunk 1]
```diff
@@ -57,29 +48,3 @@ namespace :replay_submission do
     end
   end
 end
-
-namespace :replay_hmcts_adapter_submission do
```
Comment (StevenLeighton21) [inline lib/tasks/replay_submission.rake:61]: I think we still want this? Unless it's functionally no different from the replay submission job, as it's used to replay submissions to the HMCTS adapter I think

Comment (zheileman) [reply lib/tasks/replay_submission.rake:61]: I assumed it does the same (maybe in a different way?) to the existing `rake replay_submission:process[<submission_id>]`

The existing task uses the `Delayed::Job` table, and the hmcts one used the `Submission` table. But I don't know the reason behind this.

I can leave the deleted task if we are not sure.

Comment (zheileman) [reply lib/tasks/replay_submission.rake:61]: I think I'm going to leave the task, based on the background info in this original PR
https://github.com/ministryofjustice/fb-submitter/pull/440

**Review by StevenLeighton21 — APPROVED** [review]

**Pull request merged**